rust/datafusion/src/physical_plan/group_scalar.rs

```diff
@@ -22,10 +22,12 @@ use std::convert::{From, TryFrom};
 
 use crate::error::{DataFusionError, Result};
 use crate::scalar::ScalarValue;
+use arrow::datatypes::Decimal128Type;
 
 /// Enumeration of types that can be used in a GROUP BY expression
 #[derive(Debug, PartialEq, Eq, Hash, Clone)]
 pub(crate) enum GroupByScalar {
+    Decimal128(i128, usize, usize),
```

Comment:
@alamb Should I store it as (i128, usize, usize) or `Decimal128Type` (which is a structure)?

Reply:
Change it to 

```
    // Box is needed to reduce memory usage
    Decimal128(Box<Decimal128Type>),
```

Reply:
I am not sure @ovr  -- I have sort of lost the context of this PR in my head. When you are ready let me know and I can find time to review this again.

Just to be super clear, your goal is to implement the Decimal128 and Decimal256 types that are defined in the Arrow spec, right? Not an extension?

Reply:
> your goal is to implement the Decimal128 and Decimal256 types

right. In this PR only `Decimal128 `, `Decimal256 ` will be in next PR

Reply:
Awesome -- thank you 👍